hydro_deploy/core/src/azure.rs

```diff
@@ -126,12 +127,8 @@ impl Host for AzureHost {
         self
     }
 
-    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
-        self
-    }
-
     fn collect_resources(&self, resource_batch: &mut ResourceBatch) {
-        if self.launched.is_some() {
+        if self.launched.get().is_some() {
             return;
         }
```

Comment:
Note that there could be a race condition here now - it is now up to the user to ensure that `collect_resources()` and `provision()` are not called at the same time from different threads. I'm not sure what `collect_resources()` does, it seems to need to be called before `provision()`?

Reply:
Yes, `collect_resources` always needs to be called before `provision`. What `collect_resources` does is dump declarations for the cloud resources that are required into a global Terraform config. Then, Hydro Deploy executes that config, and then `provision` establishes a connection to the created machines.